modules/bastion/main.tf

```diff
@@ -34,20 +32,11 @@ locals {
   })
 }
 
-data "aws_route53_zone" "vanity" {
-  count = local.route53_enabled ? 1 : 0
+data "aws_route53_zone" "route53_zone" {
+  count = try(length(var.vanity_domain), 0) > 0 ? 1 : 0
   name  = var.vanity_domain
 }
 
-resource "aws_route53_record" "default" {
-  count   = local.route53_enabled ? 1 : 0
-  zone_id = join("", data.aws_route53_zone.vanity.*.zone_id)
-  name    = "${var.custom_bastion_hostname}.${var.vanity_domain}"
-  type    = "A"
-  ttl     = "300"
-  records = aws_eip.static.*.public_ip
-}
-
 module "aws_key_pair" {
```

Comment:
With ssm, the key pair is not needed. It would be nice to disable this as a default and only enabled if explicitly needed. Can be done in a separate pr.

Reply:
There is a definite use case for using the key pair, as SSM is toggleable and hence in some cases a traditional `sshd` bastion will be used in lieu of an SSM bastion, but I will disable this module by default

Reply:
resolved